server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -68,6 +68,7 @@ public class ShardLimitValidator {
         "cluster.max_shards_per_node",
         1000,
         1,
+        1_500_000,
```

Comment:
This can also result in int overflow when executing
```
Math.min(maxShardsInCluster, maxShardsPerNodeSetting * nodeCount);
``` in ShardLimitValidator.java as it depends on nodeCount

Make the change here and handle the multiplication in long and check if it is exceed Integer.MAX_VALUE
```
        if (maxShardsInCluster == -1) {
            maxShardsInCluster = maxShardsPerNodeSetting * nodeCount;
        } else {
            maxShardsInCluster = Math.min(maxShardsInCluster, maxShardsPerNodeSetting * nodeCount);
        }
```